File: cli/tests/golden/artifacts/csharp/customer.CustomerRequest.v1.cs
#nullable enable
using System;
using System.Collections.Generic;

namespace Modelable.Customer;

public sealed record CustomerCustomerRequestV1
{
    public required Guid CustomerId { get; init; }
    public required string DisplayName { get; init; }
    public required string Email { get; init; }
    public required string Status { get; init; }
    public required List<string> Tags { get; init; }
    public required Dictionary<string, int> Metadata { get; init; }
    public CustomerCustomerRequestV1Address? Address { get; init; }
    public string? FavoriteProduct { get; init; }
}

public sealed record CustomerCustomerRequestV1Address
{
    public required string Line1 { get; init; }
    public string? Line2 { get; init; }
}

File: cli/tests/golden/artifacts/typescript/customer.CustomerRequest.v1.ts
/**
 * @modelable domain: customer
 * @modelable name: CustomerRequest
 * @modelable owner: customer-team
 * @modelable kind: projection
 * @modelable version: 1
 * @modelable source: customer.Customer@1
 */
import type { CatalogProductV1 } from "./catalog.Product.v1";

export interface CustomerCustomerRequestV1 {
  customerId: string;
  displayName: string;
  email: string;
  status: 'active' | 'suspended' | 'deleted';
  tags: string[];
  metadata: Record<string, number>;
  address?: { line1: string; line2?: string };
  favoriteProduct?: CatalogProductV1;
}
export type CustomerRequest = CustomerCustomerRequestV1;
